### PR TITLE
Rough draft for InputExtensions property

### PR DESCRIPTION
**Pull request by Shane32**
See:
- #2549 

Needs:
- XML comments
- Tests updated

### CODE REVIEW

File: src/GraphQL/Execution/ExecutionContext.cs [hunk 1]
```diff
@@ -59,6 +59,8 @@ public class ExecutionContext : IExecutionContext, IExecutionArrayPool, IDisposa
         public Dictionary<string, object?> Extensions { get; set; }
 #pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
 
+        public Inputs? InputExtensions { get; set; }
```
Comment (sungam3r) [inline src/GraphQL/Execution/ExecutionContext.cs:62]: OK. I suggest to rename `Extensions` to `OutputExtensions` as well.

File: src/GraphQL/Execution/ExecutionOptions.cs [hunk 2]
```diff
@@ -73,5 +73,7 @@ public class ExecutionOptions : IProvideUserContext
         /// from your dependency injection framework.
         /// </summary>
         public IServiceProvider? RequestServices { get; set; }
+
+        public Inputs? InputExtensions { get; set; }
```
Comment (sungam3r) [inline src/GraphQL/Execution/ExecutionOptions.cs:77]: See  https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#request-parameters:

query -> Query [OK]
operationName -> OperationName [OK]
variables -> Inputs [rename to Variables]
extensions -> InputExtensions [rename to Extensions]

?

Comment (sungam3r) [reply src/GraphQL/Execution/ExecutionOptions.cs:77]: `Inputs` class should not be renamed, only properties

Comment (Shane32) [reply src/GraphQL/Execution/ExecutionOptions.cs:77]: i agree with those suggestions

**Comment by sungam3r** [comment]
Server project is already ready for this PR:
https://github.com/graphql-dotnet/server/blob/master/src/Core/GraphQLRequest.cs#L51

File: src/GraphQL/Execution/ExecutionOptions.cs (hunk 2 above)
Comment (Shane32) [reply src/GraphQL/Execution/ExecutionOptions.cs:77]: just for clairifcation, `ExecutionOptions` will have just `Extensions`, but `IResolveFieldContext` will have `InputExtensions` and `OutputExtensions`?  Seems maybe we should keep `InputExtensions` named the same throughout.

Comment (sungam3r) [reply src/GraphQL/Execution/ExecutionOptions.cs:77]: Yes, `IResolveFieldContext` will have `InputExtensions` and `OutputExtensions`. `ExecutionOptions` essentially is an "input configuration", only input.  I would keep naming of properties closed to spec - `Extensions` for `ExecutionOptions`. Context already implies "input" interpretation.

**Comment by Shane32** [comment]
Replaced by #2766

**Pull request closed without merging**